app/src/main/java/com/dongyang/android/youdongknowme/ui/view/main/MainViewModel.kt

```diff
@@ -1,7 +1,84 @@
 package com.dongyang.android.youdongknowme.ui.view.main
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.dongyang.android.youdongknowme.data.remote.entity.Token
+import com.dongyang.android.youdongknowme.data.repository.MainRepository
 import com.dongyang.android.youdongknowme.standard.base.BaseViewModel
+import com.dongyang.android.youdongknowme.standard.network.NetworkResult
+import com.dongyang.android.youdongknowme.ui.view.util.Event
+import com.google.firebase.messaging.FirebaseMessaging
+import kotlinx.coroutines.launch
 
-class MainViewModel : BaseViewModel() {
-    // NOT IMPLEMENTS
+class MainViewModel(private val mainRepository: MainRepository) : BaseViewModel() {
+    private val _errorState: MutableLiveData<Event<Int>> = MutableLiveData()
+    val errorState: LiveData<Event<Int>> = _errorState
+
+    private val _isLoading: MutableLiveData<Boolean> = MutableLiveData()
+    val isLoading: LiveData<Boolean> = _isLoading
+
+    private val _isError: MutableLiveData<Boolean> = MutableLiveData()
+    val isError: LiveData<Boolean> = _isError
+
+    private val _myDepartment: MutableLiveData<String> = MutableLiveData()
+    val myDepartment: LiveData<String> get() = _myDepartment
+
+    private val _myTopics: MutableLiveData<List<String>> = MutableLiveData()
+    val myTopics: LiveData<List<String>> get() = _myTopics
+
+    private val _FCMToken: MutableLiveData<String> = MutableLiveData()
+    val FCMToken: LiveData<String> get() = _FCMToken
+
+    private val _isFirstLaunch: MutableLiveData<Boolean> = MutableLiveData(false)
+    val isFirstLaunch: LiveData<Boolean> get() = _isFirstLaunch
+
+    fun checkFirstLaunch() {
+        if (mainRepository.getIsFirstLaunch()) {
+            _isFirstLaunch.value = true
+        }
+    }
+
+    fun issuedToken() {
+        FirebaseMessaging.getInstance().token.addOnCompleteListener { task ->
+            if (task.isSuccessful) {
+                _FCMToken.value = task.result
+                setInitToken()
+            } else {
+                _isError.value = true
+            }
+        }
+    }
+
+    fun setInitToken() {
+        _isLoading.postValue(true)
+
+        val department = mainRepository.getUserDepartment()
+        _myDepartment.postValue(department)
+
+        viewModelScope.launch {
+            val keyword = mainRepository.getUserTopic()
+            _myTopics.value = keyword
+
+            val token = Token(
+                token = FCMToken.value.toString(),
+                department = myDepartment.value ?: "",
+                topics = myTopics.value ?: emptyList()
+            )
+
+            when (val result = mainRepository.setUserToken(token)) {
```

Comment:
```suggestion
            when (val result = mainRepository.setUserToken(
            Token(
                token = FCMToken.value.toString(),
                department = myDepartment.value ?: "",
                topics = myTopics.value ?: emptyList()) {
```
요 코드는 어떻게 생각하시는지두 궁금해여!

Reply:
`toekn`값이 들어가기 전 로직을 확인하기에는 기존 코드가 더 낫다고 판단하여 넣었지만 확인이 편리하다는 점 외에는 쓸데가 없어서 이게 더 나을거 같네요 수정하도록 하겠습니다